waku/protocol/v1/waku_mail.nim

```diff
@@ -76,8 +76,11 @@ proc requestMail*(node: EthereumNode, peerId: NodeId, request: MailRequest,
     return result
 
 proc p2pRequestHandler(peer: Peer, envelope: Envelope) =
-  # Mail server p2p request implementation
-  discard
+    let decoded = decode(envelope.data)
```

Comment:
This is allowed, but if IIRC the current status-go mailservers encrypt the data with a (hardcoded??) symmetric key. 
So fine for testing I guess, but you'll have to add the option to pass a key to decrypt with. (The decode call allows passing symmetric or asymmetric key: https://github.com/status-im/nim-eth/blob/master/eth/p2p/rlpx_protocols/whisper/whisper_types.nim#L317)